presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkPhysicalResourceCalculator.java

```diff
@@ -34,29 +38,70 @@
+        if (!noAllocationStrategyEnabled(session)) {
+            log.info(String.format("ResourceAllocationStrategy disabled. Executing query %s with %s", session.getQueryId(), defaultResourceSettings));
+            return defaultResourceSettings;
+        }
 
+        double inputDataInBytes = prestoSparkSourceStatsCollector.collectSourceStats(plan);
+        DataSize inputSize = new DataSize(inputDataInBytes, BYTE);
         if (inputDataInBytes < 0) {
-            log.warn(String.format("Input data statistics missing, inputDataInBytes=%.2f skipping automatic resource tuning.", inputDataInBytes));
-            return DISABLED_PHYSICAL_RESOURCE_SETTING;
+            log.warn(String.format("Input data statistics missing, inputDataInBytes=%.2f skipping automatic resource tuning. Executing query %s with %s",
+                    inputDataInBytes, session.getQueryId(), defaultResourceSettings));
+            return defaultResourceSettings;
+        }
+        else if (Double.isNaN(inputDataInBytes)) {
+            log.warn(String.format("Failed to retrieve correct size, inputDataInBytes=%.2f skipping automatic resource tuning. Executing query %s with %s",
+                    inputDataInBytes, session.getQueryId(), defaultResourceSettings));
+            return defaultResourceSettings;
+        }
+        // update hashPartitionCount only if resource allocation or hash partition allocation is enabled
+        if (isSparkResourceAllocationStrategyEnabled(session) || isSparkHashPartitionCountAllocationStrategyEnabled(session)) {
+            hashPartitionCount = calculateHashPartitionCount(session, inputSize);
         }
 
-        if ((inputDataInBytes > Double.MAX_VALUE) || (Double.isNaN(inputDataInBytes))) {
-            log.warn(String.format("Failed to retrieve correct size, data read=%.2f, skipping automatic resource tuning.", inputDataInBytes));
-            return DISABLED_PHYSICAL_RESOURCE_SETTING;
+        // update maxExecutorCount only if resource allocation or executor allocation is enabled
+        if (isSparkResourceAllocationStrategyEnabled(session) || isSparkExecutorAllocationStrategyEnabled(session)) {
+            maxExecutorCount = Optional.of(calculateExecutorCount(session, inputSize));
         }
+        PhysicalResourceSettings finalResourceSettings = new PhysicalResourceSettings(hashPartitionCount, maxExecutorCount);
 
-        DataSize inputSize = new DataSize(inputDataInBytes, BYTE);
-        int executorCount = calculateExecutorCount(session, inputSize);
-        int hashPartitionCount = calculateHashPartitionCount(session, inputSize);
+        log.info(String.format("Executing query %s with %s based on resource allocation strategy", session.getQueryId(), finalResourceSettings));
+        return finalResourceSettings;
+    }
 
-        return new PhysicalResourceSettings(executorCount, hashPartitionCount);
+    private static boolean noAllocationStrategyEnabled(Session session)
```

Comment:
if the checks are `OR` ed , then the method name should be `anyAllocationStrategyEnabled`

Reply:
Oops. Fixed.